### ISSUE TITLE
[2019 ICLR] Learning What and Where to Attend

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#47]
본 논문은 ClickMe dataset과 GALA 모듈을 제안함

논문의 주요 골자는 다음과 같음
1. ClickMe dataset에는 사람의 image에 대한 attention map data를 가지고 있는데, 이 attention map을 이용하여 모델을 학습하면 성능 향상
2. GALA module은 global-and-local attention을 가지기에 성능 향상

논문 내용 중 ClickMe dataset, GALA, experiment만 간단하게 설명

## 1. ClickMe dataset
![image](https://user-images.githubusercontent.com/49643709/148179745-d1159fa2-8993-44b1-9037-d8121f255be3.png)
![image](https://user-images.githubusercontent.com/49643709/148180383-4e952f50-7de2-4f7d-8eae-ba0fc2680259.png)

CNN은 figure-ground mechanism이 없어 context와 foreground object를 single perceptual unit으로 묶도록 강요되기에 representation이 사람이 보는 것과 달리 더 분산됨
(In the absence of figure-ground mechanisms, DCNs are compelled to associate foreground objects and their context as single perceptual units. This leads to DCN representations that are significantly more distributed compared to those used by humans)

근래에 와서 attention을 이용해서 해결하려는 노력들이 보이지만, 이는 사람의 attention과는 다소 다름

이를 해결하기 위해 저자들은 ImageNet의 subset에 대해 사람이 어떠한 부분을 보고 구분하는지에 대한, 즉 image에 대한 사람의 attention map dataset을 구축하고 이를 ClickMe dataset라고 부름

Animal categories의 경우 대부분의 경우 facial component이며, inanimate object의 경우 front-oriented bias와 더불어 distinguishing part이었음 (engine, cockpit, wheel, ...)

자세한 구축 방법 및 dataset의 statistics, reliability 등은 논문 참고

### 2.1. Intuition
2 types of feature
- Global feature : capture contextual information which guides visual processing to task-relevant image locations
- Local feature : drive separate form of attention known as visual-saliency

2 types of attention
- Channel-wise attention : learn a task-specific feature modulation
- Spatial attention : learn a spatial mask that enhance/suppress the activity of computational units inside/outside an attention spotlight 

기존의 SENet에서 제시한 SE block의 경우, channel-wise attention만 사용하여 global feature만 봤다면, GALA의 경우 spatial attention을 통해 local feature도 볼 수 있게 만듬

```
요즘 논문이 아니다 보니 spatial attention이 왜 local feature인지에 대한 의문이 들 수 있음
이 당시의 spatial attention은 `Attention is all you need`에서 제시된 self-attention이 아님
어떤 이미지가 가지고 있는 saliency에 대해 어떤 것들을 중요하게 여길 지에 대한 것임
이러한 측면에서 local feature라 표현한다고 이해하고 넘어가길...

참고로 `Attention is all you need`에 나오는 self-attention은 non-local operation으로 global feature임
```

### 2.2. Architecture
![image](https://user-images.githubusercontent.com/49643709/148180572-ba71c21f-6f52-44af-9d13-77c0dfb3bcc5.png)

#### 2.2.1. Global Attention

SENet의 SE block과 같음

1. Global average pool : calculate per-channel 'summary statistics'
2. FC (shrink) + ReLU + FC (expand) : apply non-linear transformation to these statistics, enable to learn complex dependencies between channels (reduction ratio r = 4)

#### 2.2.2. Local Feature Attention
- Conv (shrink) + ReLU + Conv (collapse) : conv with kernel size 1 * 1 (reduction ratio r = 4)

#### 2.2.3. Integration
![image](https://user-images.githubusercontent.com/49643709/148313756-0d3ef025-e456-4e69-9815-541240375b3a.png)

- Combination of additive and multiplicative
- Selection of tanh : gives additional ability to dis-inhibit bottom-up feature activations, flip the signs of its individual unit responses

#### 2.2.4. Implementation
- ResNet-50에서 24, 27, 30, 33, 36, 39 layer에 적용
- Residual module 중 skip path가 아닌 residual path의 가장 마지막에 적용 (SENet의 SE block처럼)

### 2.3. Co-training GALA with ClickMe
![image](https://user-images.githubusercontent.com/49643709/148378695-b3a11d10-48a5-4a97-bcce-3c446f0f6c57.png)

저자들이 만든 ClickMe의 image에 대한 attention map을 이용하여 GALA block output에 supervision을 줌

즉 GALA block을 통해 나온 attention map이 ClickMe의 attention map와 같게 loss function을 설계하여 학습

이 때 ClickMe의 attention map과 GALA block의 attention shape가 안맞음

이를 위해 ClickMe attention map을 GALA attention에 맞춰 bicubic interpolation (width, height shape가 맞춰짐)

GALA attention map을 column-wise L2 norm을 이용하여 channel을 1로 만듬 (channel shape가 맞춰짐)

Lambda는 hyperparameter로 6을 사용함

### 3.1. Trade-off between recognition performance and ClickMe map prediction
![image](https://user-images.githubusercontent.com/49643709/148379739-e2ae190b-334e-41aa-aa01-473e4a58f9fa.png)

- Lambda가 6일 때 최적임

### 3.2. ImageNet dataset
![image](https://user-images.githubusercontent.com/49643709/148380870-ef100434-5a2b-4564-a1cf-5ab561608915.png)

- GALA module, ClickMe map을 사용하면 성능 향상
- Coarse bounding box를 사용하는 것보다 fine-grained ClickMe attention map이 효과가 더 좋음

### 3.3. Model interpretability
![image](https://user-images.githubusercontent.com/49643709/148381223-aa8f8482-5b89-4e75-bba5-44c00f3dd4b7.png)
